onedal/cluster/kmeans.py

```diff
@@ -144,58 +136,86 @@ def _check_params_vs_input(
             self._n_init = 1
         assert self.algorithm == "lloyd"
 
-    def _get_onedal_params(self, dtype=np.float32):
+    def _get_onedal_params(self, is_csr=False, dtype=np.float32, result_options=None):
         thr = self._tol if hasattr(self, "_tol") else self.tol
         return {
             "fptype": "float" if dtype == np.float32 else "double",
-            "method": "by_default",
+            "method": "lloyd_csr" if is_csr else "by_default",
```

Comment:
Do we need to add version check for this param?

Reply:
version check is added in sklearnex, will fall back to stock for earlier oneDAL version